checker/src/z3_solver.rs

```diff
@@ -1139,8 +1139,12 @@ impl Z3Solver {
             Expression::TaggedExpression { operand, .. } => {
                 self.get_as_numeric_z3_ast(&operand.expression)
             }
-            Expression::Transmute { operand, .. } => {
-                self.get_as_numeric_z3_ast(&operand.expression)
+            Expression::Transmute { operand, target_type } => {
+                if target_type.is_integer() || target_type.is_floating_point_number() {
```

Comment:
numeric_cast has logic for target_type == ExpressionType::Char, so it probably makes sense to use it for that case as well. I can imagine that transmuting a bit vector to a char might just happen somewhere, someday.

Reply:
Are you implying that simply dispatching to `numeric_cast` would be enough in the case of `Transmute`, without the conditional to fall back to `get_as_numeric_z3_ast`? I can see that it will fall back to `get_as_numeric_z3_ast` anyways. If you confirm, I will make a new commit.

Reply:
You'll still need the fall back. The idea would be to also use numeric_cast only in the case where target_type == ExpressionType::Char. I imagine that it might be interesting to write a test case for this.

Reply:
But if I add `Char` type test like:
```rs
            Expression::Transmute { operand, target_type } => {
                if target_type.is_integer() || target_type.is_floating_point_number() || target_type == ExpressionType::Char {
                    self.numeric_cast(&operand.expression, *target_type)
                } else {
                    self.get_as_numeric_z3_ast(&operand.expression)
                }
            }
```
it would be the same as
```rs
Expression::Transmute { operand, target_type } => self.numeric_cast(&operand.expression, *target_type),
```
because `numeric_cast` basically falls back to calling `get_as_numeric_z3_ast` internally.

Or perhaps I misunderstood your intent!

Reply:
Never mind then. I'll look into this separately when I get a chance.